### PR TITLE
[1] feat(937): Add config pipeline fields and scmUrl schema

### PR DESCRIPTION
**Pull request by Filbird**
**Update 5/30**

Add `configPipelineSha` field to the events model. 

---
**Update 5/29**

Replace `isConfigPipeline` field with `scmUrls`.

---

Add optional fields `configPipelineId` and `isConfigPipeline` to the pipeline model to distinguish between **config**, **child**, and **regular** pipelines (unsure how to officially name each of these types of pipelines).

`configPipelineId` is the id of the pipeline that contains the configuration for this pipeline. This field will only be populated for child pipelines.

`isConfigPipeline` is true if the pipeline is a config pipeline.

We should discuss if adding these fields is the correct approach to distinguish between different pipeline types.

---

Also add `scmUrls ` schema into the pipeline level config.

https://github.com/screwdriver-cd/screwdriver/issues/937

### CODE REVIEW

File: models/pipeline.js [hunk 1]
```diff
@@ -52,7 +52,15 @@ const MODEL = {
...
     lastEventId: Joi.number().integer().positive()
         .description('Identifier of last event')
-        .example(123345)
+        .example(123345),
+
+    configPipelineId: Joi.number().integer().positive()
+        .description('Identifier of pipeline containing external configuration')
+        .example(123),
+
+    isConfigPipeline: Joi.boolean()
```
Comment (jithin1987) [inline models/pipeline.js]: can use use presence of `scmUrls` to infer this ?

Comment (Filbird) [reply models/pipeline.js]: yep, makes sense!

Comment (minz1027) [reply models/pipeline.js]: Then need to add `scmUrls` to this pipeline model @

File: models/event.js [hunk 2]
```diff
@@ -44,6 +44,10 @@ const MODEL = {
         .string().hex().length(40)
         .description('SHA this project was built on')
         .example('ccc49349d3cffbd12ea9e3d41521480b4aa5de5f'),
+    configPipelineSha: Joi
+        .string().hex().length(40)
+        .description('SHA of the configurgation pipeline this project depends on')
```
Comment (tkyi) [inline models/event.js]: configuration*

**Pull request merged**